query-string/query-string.d.ts

```diff
@@ -4,19 +4,21 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare module "query-string" {
+    type primitive = string | boolean | number;
+
     /**
      * Parse a query string into an object.
      * Leading ? or # are ignored, so you can pass location.search or location.hash directly.
      * @param str
      */
-    export function parse(str: string): any;
+    export function parse(str: string): {[key: string]: string | string[]};
 
     /**
      * Stringify an object into a query string, sorting the keys.
      *
      * @param obj
      */
-    export function stringify(obj: any, options?: {strict: boolean}): string;
+    export function stringify(obj: {[key: string]: primitive | primitive[]}, options?: { strict: boolean }): string;
```

Comment:
There is also a new option [encode](https://github.com/sindresorhus/query-string#encode).